### PR TITLE
8269761: idea.sh missing .exe suffix when invoking javac on WSL

### PR DESCRIPTION
**Pull request by JornVernee**
From the JBS issue:

At the end, idea.sh tries to invoke javac, but when running on WSL this results in the following error:

    bin/idea.sh: line 249: /mnt/c/progra~1/java/jdk-16/bin/javac: No such file or directory

Adding a .exe suffix to the javac path fixes this issue, which can be done just for WSL.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269761](https://bugs.openjdk.java.net/browse/JDK-8269761): idea.sh missing .exe suffix when invoking javac on WSL

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - **Reviewer**)
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4656/head:pull/4656` \
`$ git checkout pull/4656`

Update a local copy of the PR: \
`$ git checkout pull/4656` \
`$ git pull https://git.openjdk.java.net/jdk pull/4656/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4656`

View PR using the GUI difftool: \
`$ git pr show -t 4656`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4656.diff">https://git.openjdk.java.net/jdk/pull/4656.diff</a>

</details>

### CODE REVIEW

**Review by mcimadamore — APPROVED** [review]
Nice! Thanks.

**Comment by mcimadamore** [comment]
Question though: with WSL, the boot JDK could also be a Linux JDK, no? If that's the case, then the .exe suffix would be bad?

**Comment by erikj79** [comment]
> Question though: with WSL, the boot JDK could also be a Linux JDK, no? If that's the case, then the .exe suffix would be bad?

That's a good point, maybe check if BOOT_JDK/bin/java.exe exists and have that as the condition?

**Comment by JornVernee** [comment]
I realized this as well, but it doesn't look like the script works when using a linux boot JDK any way currently. The source file that is passed to `javac` is a Windows path:

```
JAVAC_SOURCE_FILE=`$CYGPATH -am $IDEA_OUTPUT/src/idea/IdeaLoggerWrapper.java`
...
$BOOT_JDK/bin/$JAVAC$EXE_SUFFIX -d $JAVAC_CLASSES -sourcepath $JAVAC_SOURCE_PATH -cp $JAVAC_CP $JAVAC_SOURCE_FILE
```

So, I could change the test to look for `BOOT_JDK/bin/java.exe`, but then the script fails with:

```
error: file not found: H:/openjdk/git-jdk2/.idea/src/idea/IdeaLoggerWrapper.java
```

The rest of the script checks for `"x$WSL_DISTRO_NAME" != "x"` and then uses Windows style paths. So, I'd rather leave the current check for adding the `.exe` suffix to do the same (consistent with the rest of the script), and leave adding support for running the script with a linux boot JDK on WSL for another time, since that doesn't seem to be a quick fix.

**Comment by JornVernee** [comment]
Hmm, now that I'm looking again, I see that an earlier block also tries to set `JAVAC` to `javac.exe` on WSL, but it never gets executed because `CYGPATH` is set in my environment:

```
if [ "x$CYGPATH" != "x" ] ; then ## CYGPATH may be set in env.cfg
  ...
  JAVAC=javac
elif [ "x$WSL_DISTRO_NAME" != "x" ]; then
  ...
  JAVAC=javac.exe
else
```

Of course, when I remove my fix, and move the `"x$WSL_DISTRO_NAME" != "x"` to the front, this also fails on a linux boot JDK because `javac.exe` doesn't exist. If I remove the `.exe` suffix then the usage of `realpath` in that same block makes the script work on a linux boot JDK, but fails again on a Windows boot JDK because of the missing .exe suffix.

I'll try to see if I can make the existing logic work on both JDK types

**Review by mcimadamore — APPROVED** [review]
Looks good!

File: bin/idea.sh [hunk 1]
```diff
@@ -69,6 +69,9 @@ do
   shift
 done
 
+if [ -e $IDEA_OUTPUT ] ; then
+    rm -r $IDEA_OUTPUT
+fi
```
Comment (JornVernee) [inline bin/idea.sh:74]: Was getting some weird errors when the `.idea` folder already existed when running the script, so I added this. It seems useful, so I've left it in.

Comment (mcimadamore) [reply bin/idea.sh:74]: Yep - noticed that, it's good

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**